rust/sbp/src/sbp_tools.rs

```diff
@@ -1,13 +1,52 @@
+use std::{collections::HashMap, convert::TryFrom, fmt::Debug};
+
 #[cfg(feature = "swiftnav-rs")]
 use swiftnav_rs::time::GpsTime;
 
 #[cfg(feature = "swiftnav-rs")]
-use crate::{
-    messages::SBPMessage,
-    time::{GpsTimeError, MessageTime, RoverTime},
-};
+use crate::time::{GpsTimeError, MessageTime, RoverTime};
+
+use crate::messages::{SBPMessage, SBP};
+
+pub struct Dispatcher {
+    callbacks: HashMap<u16, Vec<Box<dyn FnMut(SBP)>>>,
+}
+
+impl Dispatcher {
+    pub fn new() -> Self {
+        Self {
+            callbacks: HashMap::new(),
+        }
+    }
+
+    pub fn on<F, M>(&mut self, mut func: F)
```

Comment:
there should also be a method `on_once` where the callback gets removed after it gets triggered once